src/Gaufrette/Adapter/Ftp.php

```diff
@@ -77,7 +77,7 @@ public function write($key, $content)
         $this->ensureDirectoryExists($this->directory, $this->create);
 
         $path = $this->computePath($key);
-        $directory = str_replace('\\', '/', dirname($path));
+        $directory = str_replace('\\', '/', \Gaufrette\Util\Path::dirname($path));
```

Comment:
no need to use ``str_replace`` here anymore, as ``\Gaufrette\Util\Path::dirname`` is doing it already